src/laserscan_multi_merger.cpp

```diff
@@ -152,70 +152,87 @@ void LaserscanMerger::scanCallback(const sensor_msgs::LaserScan::ConstPtr& scan,
+		pointcloud_to_laserscan(&merged_cloud);
 	}
 }
 
-void LaserscanMerger::pointcloud_to_laserscan(Eigen::MatrixXf points, pcl::PCLPointCloud2 *merged_cloud)
+void LaserscanMerger::pointcloud_to_laserscan(pcl::PCLPointCloud2 *merged_cloud)
 {
-	sensor_msgs::LaserScanPtr output(new sensor_msgs::LaserScan());
-	output->header = pcl_conversions::fromPCL(merged_cloud->header);
-	output->header.frame_id = destination_frame.c_str();
-	output->header.stamp = ros::Time::now();  //fixes #265
-	output->angle_min = this->angle_min;
-	output->angle_max = this->angle_max;
-	output->angle_increment = this->angle_increment;
-	output->time_increment = this->time_increment;
-	output->scan_time = this->scan_time;
-	output->range_min = this->range_min;
-	output->range_max = this->range_max;
-
-	uint32_t ranges_size = std::ceil((output->angle_max - output->angle_min) / output->angle_increment);
+    sensor_msgs::LaserScanPtr output(new sensor_msgs::LaserScan());
+    output->header = pcl_conversions::fromPCL(merged_cloud->header);
+    output->header.frame_id = destination_frame.c_str();
+    output->header.stamp = ros::Time::now();
+    output->angle_min = this->angle_min;
+    output->angle_max = this->angle_max;
+    output->angle_increment = this->angle_increment;
+    output->time_increment = this->time_increment;
+    output->scan_time = this->scan_time;
+    output->range_min = this->range_min;
+    output->range_max = this->range_max;
+
+    std::vector<float> intensities;
+
+    uint32_t ranges_size = std::ceil((output->angle_max - output->angle_min) / output->angle_increment);
 	if (set_inf_to_the_points_exceed_range_max)
 	{
-		output->ranges.assign(ranges_size, std::numeric_limits<float>::infinity());
+    	output->ranges.assign(ranges_size, std::numeric_limits<float>::infinity());
 	}
 	else
 	{
 		output->ranges.assign(ranges_size, output->range_max + 1.0);
 	}
 
-	for(int i=0; i<points.cols(); i++)
-	{
-		const float &x = points(0,i);
-		const float &y = points(1,i);
-		const float &z = points(2,i);
-
-		if ( std::isnan(x) || std::isnan(y) || std::isnan(z) )
-		{
-			ROS_DEBUG("rejected for nan in point(%f, %f, %f)\n", x, y, z);
-			continue;
-		}
-
-		double range_sq = y*y+x*x;
-		double range_min_sq_ = output->range_min * output->range_min;
-		if (range_sq < range_min_sq_) {
-			ROS_DEBUG("rejected for range %f below minimum value %f. Point: (%f, %f, %f)", range_sq, range_min_sq_, x, y, z);
-			continue;
-		}
-
-		double angle = atan2(y, x);
-		if (angle < output->angle_min || angle > output->angle_max)
-		{
-			ROS_DEBUG("rejected for angle %f not in range (%f, %f)\n", angle, output->angle_min, output->angle_max);
-			continue;
-		}
-		int index = (angle - output->angle_min) / output->angle_increment;
-
+    for (size_t i = 0; i < merged_cloud->data.size(); i += merged_cloud->point_step)
+    {
+        float x, y, z;
+        float intensity_value;
+        memcpy(&x, &merged_cloud->data[i], sizeof(float));
```

Comment:
memo:.
Since `data` is `std::vector< std::uint8_t >`, `data` is copied in this way.

It could have been casted as follows.
x = static_cast<std::vector< float>)(merged_cloud->data)[i];